platform/security/rest/security-rest-clientapi/src/main/java/org/codice/ddf/cxf/client/ClientFactoryFactory.java

```diff
@@ -222,6 +223,32 @@
       String clientSecret,
       String oauthFlow);
 
+  /**
+   * Returns an initialized SecureCxfClientFactory
+   *
+   * @param endpointUrl
+   * @param interfaceClass
+   * @param sourceId
+   * @param discoveryUrl
+   * @param clientId
+   * @param clientSecret
+   * @param username
+   * @param password
+   * @param additionalParameters
+   * @param <T>
+   * @return
+   */
+  <T> SecureCxfClientFactory<T> getSecureCxfClientFactory(
```

Comment:
also these methods are getting crazy and actually quite hard to use would a builder pattern be better?

Reply:
instead of doing a refactor for this ticket would you mind just having a constructor that has all possible arguments? for our use case we have to pass in a provider which is not part of this constructor but we also need additionalParameters which is not part of the other constructors.